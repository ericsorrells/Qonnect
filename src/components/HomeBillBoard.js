// ========================================================================================
import React from 'react';
import { Link }  from 'react-router-dom';
// ========================================================================================

const HomeBillBoard = () => {
  return (
    <div className='home__container'>
      <div className='home__inner-container'>
        <div className='home__text-container'>
          <div className='home__text'>New Friends.</div>
          <div className='home__text'>New Experiences.</div>
          <div className='home__text'>Have Fun.</div>
        </div>
        <Link to='signup' className='home__button'>Have Fun Now!</Link>
      </div>
    </div>
  )
}

export default HomeBillBoard;