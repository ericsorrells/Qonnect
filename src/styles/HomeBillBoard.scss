.home__container {
  background: url('/images/home_image.jpg');   // this is to the 'public' folder
  height: 570px;
  width: auto;
  background-size: cover;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
}

.home__inner-container {
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  align-items: flex-end;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 50px;
}

.home__text-container {
  display: flex;
  flex-direction: column;
}

.home__text {
  background-color: white;
  font-size: 2.75rem;
  // width: 100%;
  margin-top: 10px;
  padding-top: 3px;
  padding-bottom: 3px;
}

.home__button {
  background-color: $main-color;
  padding: 8px 30px;
  font-size: 1.75rem;
  color: white;
  border: 1px solid $main-color;
  border-radius: 5px;
}