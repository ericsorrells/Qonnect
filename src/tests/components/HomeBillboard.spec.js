// ========================================================================================
import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import 'jest-enzyme';
import { Link } from 'react-router-dom';
// ========================================================================================
import HomeBillBoard from '../../components/HomeBillBoard';
// ========================================================================================

describe('<HomeBillBoard />', () => {
  const component = mount(
    <MemoryRouter initialEntries={[ {key: 'testKey' } ]} >
      <HomeBillBoard />
    </MemoryRouter>
  );

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('had a link to the SignUp page', () => {
    expect(component.find('a')).toHaveProp('href', 'signup')
  });
});