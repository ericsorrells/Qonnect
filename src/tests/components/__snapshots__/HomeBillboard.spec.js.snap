// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeBillBoard /> renders correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HomeBillBoard>
      <div
        className="home__container"
      >
        <div
          className="home__inner-container"
        >
          <div
            className="home__text-container"
          >
            <div
              className="home__text"
            >
              New Friends.
            </div>
            <div
              className="home__text"
            >
              New Experiences.
            </div>
            <div
              className="home__text"
            >
              Have Fun.
            </div>
          </div>
          <Link
            className="home__button"
            replace={false}
            to="signup"
          >
            <a
              className="home__button"
              href="signup"
              onClick={[Function]}
            >
              Have Fun Now!
            </a>
          </Link>
        </div>
      </div>
    </HomeBillBoard>
  </Router>
</MemoryRouter>
`;
